yt_dlp/YoutubeDL.py

```diff
@@ -2490,6 +2508,11 @@ def _calc_headers(self, info_dict):
                     values.append(f'Version={cookie.version}')
             info_dict['cookies'] = '; '.join(values)
```

Comment:
Where is this going to be used? Is it just a pass-through for write/load-info-json?

Does it need to be documented in `extractor/common.py`, or if not should the key be `_cookies`?

On reflection it looks as if a whole lot of Set-Cookie headers are being strung together. Does that make sense? Shouldn't the result be a list of items each like `'; '.join(values)`, one per cookie iteration?

Ofc above depends on what is going to be done with the values.

Reply:
> Where is this going to be used? Is it just a pass-through for write/load-info-json?

They are currently only round tripped through info json, yes.

> Does it need to be documented in `extractor/common.py`

`extractor/common.py` is not the correct place for this, since the cookies are not part of the extractor extracting stuff (similarly to how headers are not documented there). We are not documenting the property yet as it might be subject to change iirc.

> ..., or if not should the key be `_cookies`?
> Ofc above depends on what is going to be done with the values.

We did not privatize the value since we want third parties to use the value for now.

> On reflection it looks as if a whole lot of Set-Cookie headers are being strung together. Does that make sense? Shouldn't the result be a list of items each like `'; '.join(values)`, one per cookie iteration?

We are using a single `Set-Cookie` header string since `SimpleCookie` allows for parsing such a construct and we are planning on supporting the same syntax in the future using `--set-cookie`.

Reply:
Sounds good: so the logic is that each `name=...` introduces a new cookie?

Ofc there is some "tension" between "not documenting the property yet as it might be subject to change" and "we want third parties to use the value for now".

Reply:
> What happens if my site sets a cookie with name `Domain`?

We are parsing according to `Set-Cookie` spec and so setting a cookie named Domain is impossible even in browsers.

Reply:
> Ofc there is some "tension" between "not documenting the property yet as it might be subject to change" and "we want third parties to use the value for now".
 
 This is a good point. IMO it's ready to be documented, I just don't know where to do that. As Grub said, doing so in the InfoExtractor docstring doesn't make sense, since the `cookies` field shouldn't & can't be accessed or modified directly by the extractors

Reply:
>...setting a cookie named Domain is impossible even in browsers.

According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie:
>A <cookie-name> can contain any US-ASCII characters except for: control characters (ASCII characters 0 up to 31 and ASCII character 127) or separator characters (space, tab and the characters: ( ) < > @ , ; : \ " / [ ] ? = { })

I'm not aware of anything that limits <cookie-name> beyond that (other than custom and practice), but happy to be proved wrong. Maybe browsers actually reject my hypothetical cookies named `Secure` and `Domain`.

In HTTP, a separate `Set-Cookie:` header is sent for each cookie, so the first `name=value` can define the name and value of the cookie. If a whole lot of these are strung together, this isn't possible. Obvs a `--set-cookie ...` option, if implemented, could be repeated in the same way as the HTTP response header.

Reply:
at least `SimpleCookie`/`LenientSimpleCookie` will reject cookies named `Domain`/`Path`/etc, and that's what we are using to parse the `cookies` field
```py
>>> from http.cookies import SimpleCookie
>>> a = SimpleCookie('name=value')
>>> a
<SimpleCookie: name='value'>
>>> b = SimpleCookie('Domain=value')
>>> b
<SimpleCookie: >
>>> c = SimpleCookie('Path=value')
>>> c
<SimpleCookie: >
>>> d = SimpleCookie('name1=value1; Domain=.example.com; Path=/; Domain=badcookie; Domain=.badexample.com; Path=/; name3=value3; Domain=.example.com; Path=/')
>>> d
<SimpleCookie: name1='value1' name3='value3'>
```

Reply:
Indeed, the parsing tactic is what I guessed, at least in CPython 2.7 and 3.11: any key in `key=val` that isn't in a dict of well-known attributes is a new cookie name, and so no cookie can have such a name. As this hasn't been changed since 2000, the restriction that cookie names can't be well-known attribute names must be an unwritten rule that "everyone knows".

RFC2109 (which is mentioned in the code) allowed setting multiple cookies separated by `,` in a `Set-Cookie` header (and this is understood by `SimpleCookie`); however the feature is gone in RFC6265 (which along with the original Netscape 1-pager linked from it) is the current spec. Also gone is a restriction that cookie names couldn't begin with `$`: this allowed cookies to be sent with attributes as `name=value; $attr=$attr_val; ...`, which was (a) not done (b) a bad idea.